lib/cucumber/runtime/ast_tree_walker.js

```diff
@@ -288,13 +292,15 @@ function AstTreeWalker(features, supportCodeLibrary, listeners, strictMode) {
     },
 
     processStep: function processStep(step, callback) {
-      if (self.isStepUndefined(step)) {
+      if(self.isDryRun()) {
+        self.skipStep(step, callback);
+      } else if (self.isStepUndefined(step)) {
```

Comment:
dry run shouldn't skip undefined steps if we are trying to follow cucumber's [example](http://www.relishapp.com/cucumber/cucumber/docs/cli/dry-run). I'd suggest adding some feature tests for this.